backend/src/api/Controllers/TodoController.cs

```diff
@@ -1,42 +1,36 @@
 using core.Ports;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using api.Resonses;
+using api.Filters;
+using core;
 
 namespace api.Controllers;
 
 [Authorize]
+[UserIdFilter]
 [Route("/api/todo")]
 [ApiController]
-public class TodoController : ControllerBase
+public class TodoController(ITodoService service, ILogger<TodoController> logger) : ControllerBase
 {
-    private readonly ITodoService _service;
-    private readonly ILogger<TodoController> _logger;
+    private readonly ITodoService _service = service;
+    private readonly ILogger<TodoController> _logger = logger;
 
-    public TodoController(ITodoService service, ILogger<TodoController> logger)
-    {
-        _service = service;
-        _logger = logger;
-    }
+    //We are sure that this is not null because of the [UserIdFilter]
+    private Guid UserId => (Guid)HttpContext.Items["UserId"]!;
 
     [HttpGet("lists")]
     public IActionResult GetLists()
     {
-        var userId = Request.UserId();
-        _logger.LogDebug("User: {user} requested his lists", userId);
-        return userId switch
-        {
-            null => Unauthorized(),
-            var user => Ok(_service.GetLists(user)),
-        };
+        _logger.LogDebug("User: {user} requested his lists", UserId);
+        return Ok(_service.GetLists(UserId).Select(l => l.ToResponse()));
     }
 
     [HttpGet("lists/{id}")]
     public IActionResult GetList([FromRoute] int id)
     {
-        var userId = Request.UserId();
-        _logger.LogDebug("User: {user} requested his list: {id}", userId, id);
-        if (userId is null) { return Unauthorized(); }
-        var list = _service.GetList(userId, id);
+        _logger.LogDebug("User: {user} requested his list: {id}", UserId, id);
+        var list = _service.GetList(UserId, id)?.ToResponse();
         return list switch

```

Comment:
Here in this switch, we have the value with the name `user`, but it is supposed to be the `List` object, am I correct?